include/vara/Feature/Constraint.h

```diff
@@ -25,55 +25,166 @@ class Constraint {
   ConstraintKind Kind;
 };
```

Comment:
Github does not let me annotate the lines above :cry: 
Both sub hierachies have the `toString` method so should be not put it into the base class?

Add this to the public API
```cpp
[[nodiscard]] virtual std::string toString() const;
```

Reply:
you're right! but how can i mark both subclasses as abstract? (BooleanConstraint and NumericConstraint?)

Reply:
As long as they don't offer a `toString` method them selfs (or higher up the hierarchy) they are still abstract.